### DISCUSSION TITLE
What is the difference between the two options?

### DISCUSSION THREAD

**Discussion by katastrophe92** [celzero/rethink-app#1191] [Q&A]
![SmartSelect_20240114_083909_Rethink.jpg](https://github.com/celzero/rethink-app/assets/103950507/de655ff8-d7fa-4150-82bb-7044e0469dc7)

Or is better to have the block lists on the device? 

Thank you!

**Comment by ignoramous** [comment]
Some folks want to use the blocklists but with a different upstream server instead of Rethink's. In that case, local blocklists come in handy.

Also, with local blocklists you are in control and avoid any changes in Rethink's server from affecting blocking locally on your Android.

Besides, performance wise, there's not much difference between the two, if *DNS booster* is enabled.

**Comment by katastrophe92** [comment]
@ignoramous Thanks for you're reply! What is ''upstream server'' and does the On—device blocklists update itself?

**Reply by ignoramous** [reply]
> What is ''upstream server'' 

Any user-set DNS server (from within the Rethink app) is "upstream".

> does the On—device blocklists update itself?

It can periodically check for updates (disabled by default `Prompt on blocklist updates`), but won't download until user asks it to.

**Comment by katastrophe92** [comment]
Thanks for you're reply! I could use NextDNS as DNS then and RethinkDNS as firewall

**Reply by ignoramous** [reply]
Yes, you can.
